include/vctr/Expressions/Generic/TransformedByStaticCast.h

```diff
@@ -0,0 +1,67 @@
+/*
+  ==============================================================================
+    DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+
+    Copyright 2023 by sonible GmbH.
+
+    This file is part of VCTR - Versatile Container Templates Reconceptualized.
+
+    VCTR is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License version 3
+    only, as published by the Free Software Foundation.
+
+    VCTR is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License version 3 for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    version 3 along with VCTR.  If not, see <https://www.gnu.org/licenses/>.
+  ==============================================================================
+*/
+
+namespace vctr::Expressions
+{
+
+template <size_t extent, class SrcType, std::constructible_from<ValueType<SrcType>> DstValueType>
+class TransformedByStaticCast : public ExpressionTemplateBase
+{
+public:
+    using value_type = DstValueType;
+
+    VCTR_COMMON_UNARY_EXPRESSION_MEMBERS (TransformedByStaticCast, src)
+
+    VCTR_FORCEDINLINE constexpr DstValueType operator[] (size_t i) const
+    {
+        return static_cast<DstValueType> (src[i]);
+    }
+
+    VCTR_FORCEDINLINE const DstValueType* evalNextVectorOpInExpressionChain (DstValueType* dst) const
+    requires (sizeof (DstValueType) == sizeof (ValueType<SrcType>) || is::anyVctr<SrcType>) &&
+             has::evalNextVectorOpInExpressionChain<SrcType, ValueType<SrcType>> &&
+             std::is_trivially_destructible_v<ValueType<SrcType>> &&
+             std::is_trivially_constructible_v<DstValueType>
+    {
+        const auto* s = src.evalNextVectorOpInExpressionChain (static_cast<ValueType<SrcType>*> (static_cast<void*> (dst)));
```

Comment:
This is to silence compiler complaints about unrelated types? Probably worth a comment, but not sure whether I get the full picture…